lib/livebook/hubs.ex

```diff
@@ -0,0 +1,96 @@
+defmodule Livebook.Hubs do
+  @moduledoc false
+
+  alias Livebook.Storage
+  alias Livebook.Hubs.Hub
+
+  defmodule NotFoundError do
+    @moduledoc false
+
+    defexception [:id, plug_status: 404]
+
+    def message(%{id: id}) do
+      "could not find a hub matching #{inspect(id)}"
+    end
+  end
+
+  @namespace :hub
+
+  @doc """
+  Gets a list of hubs from storage.
+  """
+  @spec fetch_hubs() :: list(Hub.t())
+  def fetch_hubs do
+    Storage.current().all(@namespace)
+  end
+
+  @doc """
+  Gets one hub from storage.
+
+  Raises `NotFoundError` if the hub does not exist.
+  """
+  @spec hub_by_id!(String.t()) :: Hub.t()
+  def hub_by_id!(id) do
+    case Storage.current().fetch(@namespace, id) do
+      :error -> raise NotFoundError, id: id
+      {:ok, fields} -> struct!(Hub, fields)
+    end
+  end
+
+  @doc """
+  Checks if hub already exists.
+  """
+  @spec hub_exists?(Hub.t()) :: boolean()
+  def hub_exists?(%Hub{id: id}) do
+    case Storage.current().fetch(@namespace, id) do
+      :error -> false
+      {:ok, _} -> true
+    end
+  end
+
+  @doc """
+  Saves a new hub to the configured ones.
+  """
+  @spec save_hub(Hub.t()) :: Hub.t()
```

Comment:
The idea is that we won't save a Hub! We will actually save `Fly` and then we will get a Hub from it. :) Ping me on Slack later so we go over it. :)